From: Joe Shaw <joeshaw@novell.com>

Lucene.Net 1.9.1 contains overrides for most of the Query classes
which overrides the underlying Clone() method (which does a
MemberwiseClone()) with implementations that just return null.
LuceneCommon depends pretty heavily on the Clone() method, so null
return values break things badly.  This patch has been sent upstream
(2 Oct 2006), hopefully it will go away when we pull a new Lucene.Net
version.

Index: Search/MatchAllDocsQuery.cs
===================================================================
RCS file: /cvs/gnome/beagle/beagled/Lucene.Net/Search/MatchAllDocsQuery.cs,v
retrieving revision 1.2
diff -u -p -u -r1.2 MatchAllDocsQuery.cs
--- Search/MatchAllDocsQuery.cs	2 Oct 2006 17:09:05 -0000	1.2
+++ Search/MatchAllDocsQuery.cs	2 Oct 2006 18:14:32 -0000
@@ -193,11 +193,5 @@ namespace Lucene.Net.Search
 		{
 			return BitConverter.ToInt32(BitConverter.GetBytes(GetBoost()), 0);
 		}
-
-        // {{Aroush-1.9}} Do we need this?!
-        override public System.Object Clone()
-		{
-			return null;
-		}
 	}
 }
Index: Search/MultiPhraseQuery.cs
===================================================================
RCS file: /cvs/gnome/beagle/beagled/Lucene.Net/Search/MultiPhraseQuery.cs,v
retrieving revision 1.2
diff -u -p -u -r1.2 MultiPhraseQuery.cs
--- Search/MultiPhraseQuery.cs	2 Oct 2006 17:09:05 -0000	1.2
+++ Search/MultiPhraseQuery.cs	2 Oct 2006 18:14:32 -0000
@@ -337,11 +337,5 @@ namespace Lucene.Net.Search
 			
 			return buffer.ToString();
 		}
-
-        // {{Aroush-1.9}} Do we need this?!
-        override public System.Object Clone()
-		{
-			return null;
-		}
 	}
 }
Index: Search/PhrasePrefixQuery.cs
===================================================================
RCS file: /cvs/gnome/beagle/beagled/Lucene.Net/Search/PhrasePrefixQuery.cs,v
retrieving revision 1.4
diff -u -p -u -r1.4 PhrasePrefixQuery.cs
--- Search/PhrasePrefixQuery.cs	2 Oct 2006 17:09:06 -0000	1.4
+++ Search/PhrasePrefixQuery.cs	2 Oct 2006 18:14:32 -0000
@@ -312,10 +312,5 @@ namespace Lucene.Net.Search
 			
 			return buffer.ToString();
 		}
-        // {{Aroush-1.9}} Do we need this?!
-		override public System.Object Clone()
-		{
-			return null;
-		}
 	}
 }
Index: Search/PhraseQuery.cs
===================================================================
RCS file: /cvs/gnome/beagle/beagled/Lucene.Net/Search/PhraseQuery.cs,v
retrieving revision 1.5
diff -u -p -u -r1.5 PhraseQuery.cs
--- Search/PhraseQuery.cs	2 Oct 2006 17:09:06 -0000	1.5
+++ Search/PhraseQuery.cs	2 Oct 2006 18:14:32 -0000
@@ -336,10 +336,5 @@ namespace Lucene.Net.Search
 		{
 			return BitConverter.ToInt32(BitConverter.GetBytes(GetBoost()), 0) ^ slop ^ terms.GetHashCode() ^ positions.GetHashCode();
 		}
-		// {{Aroush-1.9}} Do we need this?!
-		override public System.Object Clone()
-		{
-			return null;
-		}
 	}
 }
Index: Search/PrefixQuery.cs
===================================================================
RCS file: /cvs/gnome/beagle/beagled/Lucene.Net/Search/PrefixQuery.cs,v
retrieving revision 1.4
diff -u -p -u -r1.4 PrefixQuery.cs
--- Search/PrefixQuery.cs	2 Oct 2006 17:09:06 -0000	1.4
+++ Search/PrefixQuery.cs	2 Oct 2006 18:14:32 -0000
@@ -104,10 +104,5 @@ namespace Lucene.Net.Search
 		{
 			return BitConverter.ToInt32(BitConverter.GetBytes(GetBoost()), 0) ^ prefix.GetHashCode();
 		}
-        // {{Aroush-1.9}} Do we need this?!
-        override public System.Object Clone()
-		{
-			return null;
-		}
 	}
 }
Index: Search/RangeQuery.cs
===================================================================
RCS file: /cvs/gnome/beagle/beagled/Lucene.Net/Search/RangeQuery.cs,v
retrieving revision 1.4
diff -u -p -u -r1.4 RangeQuery.cs
--- Search/RangeQuery.cs	2 Oct 2006 17:09:06 -0000	1.4
+++ Search/RangeQuery.cs	2 Oct 2006 18:14:32 -0000
@@ -187,10 +187,5 @@ namespace Lucene.Net.Search
 		{
 			return BitConverter.ToInt32(BitConverter.GetBytes(GetBoost()), 0) ^ (lowerTerm != null ? lowerTerm.GetHashCode():0) ^ (upperTerm != null?upperTerm.GetHashCode() : 0) ^ (this.inclusive ? 1 : 0);
 		}
-		// {{Aroush-1.9}} Do we need this?!
-		override public System.Object Clone()
-		{
-			return null;
-		}
 	}
 }
Index: Search/TermQuery.cs
===================================================================
RCS file: /cvs/gnome/beagle/beagled/Lucene.Net/Search/TermQuery.cs,v
retrieving revision 1.4
diff -u -p -u -r1.4 TermQuery.cs
--- Search/TermQuery.cs	2 Oct 2006 17:09:07 -0000	1.4
+++ Search/TermQuery.cs	2 Oct 2006 18:14:32 -0000
@@ -202,10 +202,5 @@ namespace Lucene.Net.Search
 		{
 			return BitConverter.ToInt32(BitConverter.GetBytes(GetBoost()), 0) ^ term.GetHashCode();
 		}
-		// {{Aroush-1.9}} Do we need this?!
-		public override System.Object Clone()
-		{
-			return null;
-		}
 	}
 }
